File: backend/src/main/java/com/starttostar/backend/service/StartupService.java
package com.starttostar.backend.service;

import com.starttostar.backend.entity.Startup;
import com.starttostar.backend.repository.StartupRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class StartupService {

    private final StartupRepository startupRepository;

    public StartupService(StartupRepository startupRepository) {
        this.startupRepository = startupRepository;
    }

    // CREATE
    public Startup createStartup(Startup startup) {
        return startupRepository.save(startup);
    }

    // GET ALL
    public List<Startup> getAllStartups() {
        return startupRepository.findAll();
    }

    // GET BY ID
    public Startup getStartupById(Long id) {
        return startupRepository.findById(id).orElse(null);
    }

    // UPDATE
    public Startup updateStartup(Long id, Startup updatedStartup) {

        Startup existingStartup =
                startupRepository.findById(id).orElse(null);

        if (existingStartup == null) {
            return null;
        }

        existingStartup.setStartupName(updatedStartup.getStartupName());
        existingStartup.setFounderName(updatedStartup.getFounderName());
        existingStartup.setEmail(updatedStartup.getEmail());
        existingStartup.setIndustry(updatedStartup.getIndustry());
        existingStartup.setFunding(updatedStartup.getFunding());
        existingStartup.setDescription(updatedStartup.getDescription());
        existingStartup.setWebsite(updatedStartup.getWebsite());
        if (updatedStartup.getLocation() != null) {
            existingStartup.setLocation(updatedStartup.getLocation());
        }
        if (updatedStartup.getStage() != null) {
            existingStartup.setStage(updatedStartup.getStage());
        }

        return startupRepository.save(existingStartup);
    }

    // DELETE
    public boolean deleteStartup(Long id) {
        if (startupRepository.existsById(id)) {
            startupRepository.deleteById(id);
            return true;
        }
        return false;
    }
}

File: client/src/services/StartupService.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/startups";

// GET ALL STARTUPS
export const getAllStartups = () => {
  return axios.get(API_URL);
};

// CREATE STARTUP
export const createStartup = (startup) => {
  return axios.post(API_URL, startup);
};

// GET STARTUP BY ID
export const getStartupById = (id) => {
  return axios.get(`${API_URL}/${id}`);
};

// UPDATE STARTUP
export const updateStartup = (id, startup) => {
  return axios.put(`${API_URL}/${id}`, startup);
};

// DELETE STARTUP
export const deleteStartup = (id) => {
  return axios.delete(`${API_URL}/${id}`);
};